_posts/2019-08-23-lti-and-saml-the-basics.markdown

```diff
@@ -0,0 +1,96 @@
+The authentication works through a key and secret system with some other configurable pieces on the LMS side.
+
+The terms to know in regards to authentication are the following:
+
+- **Consumer Key** - This is a shared key between the Consumer and the Tool Provider.
+- **Shared Secret** - This is the privately shared secret between the Consumer and the Tool Provider.
+- **Launch URL** - This is the URL that the user would go to when launching the Tool Provider.
+- **Domain** - This is derived from the Launch URL and seems to be optional in my experience.
+- **Privacy** - This is the level of privacy for the information shared with the Tool Provider.
+- **Custom Fields** - These are the custom fields that will allow the Tool Provider to tailor the experience to a specific Consumer.
+
+These variables are pretty simple to understand and they should be scary.
+A tough aspect is implementing the logic for authentication on the Tool Provider side.
+But fear not!
+There are libraries out there that will help like [PassportJS](http://www.passportjs.org/packages/passport-lti) and others to help.
+These libraries take care of some of the magic associated with LTI, but the give you a friendly interface to work with.
+We have been using PassportJS at work for quite a few years and we rarely have issues with our original implementation.
+I would recommend!
+
+## SAML Standard
+
+The SAML standard is more of a corporate world way to do single sign-on.
+SAML stands for Security Assertion Markup Language.
+It provides shared authentication between an Identity Provider (IDP) and a Service Provider (SP).
+This is helpful when an organization has many services they use and only want their users to have one login.
+Honestly thinking about how it works isn't too difficult, so I am going jump right in.
+
+### How it works
+
+The core of SAML is having an authoritative source that service providers can trust.
+When setting up SAML, a set of metadata is shared between the two providers to set up a secure connection and make sure that all needed values for the SP are sent.
+See the flow diagram below for the end user's flow for both SP initiated and IDP initiated.
+
+![SAML SP Diagram][]
+
+![SAML IDP Diagram][]
+
+As you can see the user begins by accessing the service provider and informing them that they belong to a certain identity provider.
+Once the service provider knows who the user is and where to send them to be authorized, they are redirected to the IDP login screen.
+After the IDP says they are good, they redirect back to the TP with a set of user identifiable information.
+When the SP sees this, it verifies that the payload was signed with the same certificate that the SP has on file.
+Now that we have proven the user is from a trusted source, the SP can give access to the user.
+
+Some terms to know are the following:
+
+- **Entry Point** - This is the endpoint on the Identity Provider side that the Service Provider redirects to for authentication.
```

Comment:
```suggestion
- **Entry Point** - This is the endpoint on the IDP side that the SP redirects to for authentication.
```
I'm indiffirent on making them shorter or not but I would expect it to be consistent.